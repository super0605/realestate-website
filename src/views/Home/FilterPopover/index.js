import React, { useState } from "react";
import { Popover } from "antd";

const FilterPopover = ({ children, filterContent: ContentComponent }) => {
  const [visible, setVisible] = useState(false);

  const handleVisibleChange = (val) => {
    setVisible(val);
  };

  function hide() {
    setVisible(false);
  }

  return (
    <Popover
      content={<ContentComponent hide={hide} />}
      placement="bottomLeft"
      trigger="click"
      visible={visible}
      onVisibleChange={handleVisibleChange}
    >
      {children}
    </Popover>
  );
};

export default FilterPopover;
