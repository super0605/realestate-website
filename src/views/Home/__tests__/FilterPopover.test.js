import * as React from "react";
import { mount, shallow } from "enzyme";
import FilterPopover from "../FilterPopover";

describe("FilterPopover", () => {
  let props;
  let mountedFilterSection;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedFilterSection) {
      mountedFilterSection = bShallow
        ? shallow(<FilterPopover {...props} />)
        : mount(<FilterPopover {...props} />);
    }
    return mountedFilterSection;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
