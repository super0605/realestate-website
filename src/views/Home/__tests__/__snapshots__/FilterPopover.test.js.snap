// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPopover should render without problems 1`] = `
<Popover
  content={
    <UNDEFINED
      hide={[Function]}
    />
  }
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  onVisibleChange={[Function]}
  overlayStyle={Object {}}
  placement="bottomLeft"
  transitionName="zoom-big"
  trigger="click"
  visible={false}
/>
`;
